### PR TITLE
#162163232 enable user rate article

### PR DESCRIPTION
**Pull request by SokoPaulSokool**
### What does this PR do?
Add rating of an article 

### Description of Task to be completed?
When a user views an article, they should be able to rate the article

### What are the relevant pivotal tracker stories?
#162163232]

### How to test the feature
- Click on the read more of any article
- The average rating of the article is displayed in form of star icons
- Click on the stars to give it a new rating

### Screenshots
image

### CODE REVIEW

**Comment by JaymesKat** [comment]
@SokoPaulSokool In the PR description, please indicate how this feature can be tested? Rename your PR heading to remove `ft`

File: src/components/articles/articleRating/RatingView.jsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import rateArticle from "../../../actions/articleActions/RateArticleActions";
+
+export class RatingView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ratings: ["", "", "", "", ""], ratingNumber: 3 };
+  }
+  componentDidMount() {
+    this.setState({ ratingNumber: this.props.article.avg_rating });
+    const ratingNumber = this.props.article.avg_rating;
```
Comment (JaymesKat) [inline src/components/articles/articleRating/RatingView.jsx]: use destructuring to  get the `avg_rating` i.e. `const {  article: { avg_rating} } = this.props`

**Comment by JaymesKat** [comment]
@SokoPaulSokool Add any missing prop-type validations

**Review by bibangamba — APPROVED** [review]

**Comment by bibangamba** [comment]
please resolve conflicts though

**Review by KicaRonaldOkello — APPROVED** [review]

**Review by Nta1e — APPROVED** [review]

**Review by bibangamba — DISMISSED** [review]
Please add functionality that asks the user to login if they are not logged in. The components and logic for this were already added in @KicaRonaldOkello and @Nta1e's PRs (which were merged) so it should be easy to incorporate. Then resolve the conflicts too. Feature works great though, good work.

**Comment by JaymesKat** [comment]
@SokoPaulSokool 
Good work however the rating component is mis-aligned. (see screenshot)
screen shot 2019-01-29 at 13 36 26

**Comment by SokoPaulSokool** [comment]
I have fixed the issue

**Review by bibangamba — APPROVED** [review]
LGTM

**Review by JaymesKat — APPROVED** [review]
@SokoPaulSokool Good work

**Pull request merged**